.github/workflows/check-generated-code.yml

```diff
@@ -0,0 +1,66 @@
+# SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+name: Check Generated Code
+
+on:
+  # run pipeline on push event of main branch
+  push:
+    branches:
+      - main
+    paths:
+      - code_generation/**
+      - .github/workflows/*
+      - power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/**
+      - power_grid_model_c/power_grid_model_c/**
+      - src/power_grid_model/core/**
```

Comment:
only the specific files that are affected are needed here. the action is not really that heavy but heavy enough.

alternatively, this entire action can also be a separate step in the `check-code-quality` action. then you don't need the specific paths, but just always run it. what do you think?

Reply:
After discussion, the action has been moved within `check-code-quality` action.